File: Resueltos_por_competidor/Gabriella_Castro/SameDifferences.java
/*
 * Autor: Gabriella Castro
 * Problema: Same Differences
 * Juez online: VJudge
 * Veredicto: Accepted
 * URL: https://vjudge.net/problem/CodeForces-1520D
 */

import java.util.*;
import java.io.*;


public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int t = Integer.parseInt(br.readLine());

		while (t-- > 0) {
			int n = Integer.parseInt(br.readLine());
			StringTokenizer st = new StringTokenizer(br.readLine());

			Map<Integer, Long> nums = new HashMap<>();

			for (int i = 1; i <= n; i++) {
				int val = Integer.parseInt(st.nextToken());
				int diff = val - i;
				nums.put(diff, nums.getOrDefault(diff, 0L) + 1);
			}

			long total = 0;

			for (long contar : nums.values()) {
				if (contar > 1) {
					total += (contar * (contar - 1)) / 2;
				}
			}

			System.out.println(total);
		}
	}
}

File: Resueltos_por_competidor/Juan_Martinez/SameDifferences.cpp
/*
 * Autor: Juan Martinez
 * Problema: Same Differences (1520D)
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/1520/D
 * Difficulty: 1200
 */
#include <bits/stdc++.h>
using namespace std;

typedef long long int ll;

int main() {
	map<int,int> par;
    ll t, n, cantPar, temp; cin>>t;
    while(t--) {
        cin>>n;
        for(int i = 0; i < n; i++) {
            cin>>temp;
            par[temp-(i+1)]++;
        }
        cantPar = 0;
        for(auto [val,cant] : par){
            if(cant > 1){
                while(cant > 0){
                    cantPar += (cant-1);
                    cant--;
                }
            }
        }
        cout<<cantPar<<endl;
        par.clear();
    }
}
